File: apps/ruby-sdk/lib/firecrawl/models/crawl_response.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Response from starting an async crawl job.
    class CrawlResponse
      attr_reader :id, :url

      def initialize(data)
        @id = data["id"]
        @url = data["url"]
      end

      def to_s
        "CrawlResponse{id=#{id}, url=#{url}}"
      end
    end
  end
end

File: apps/dot-net-sdk/Firecrawl/Models/CrawlResponse.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Response from starting an async crawl job.
/// </summary>
public class CrawlResponse
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("id")]
    public string? Id { get; set; }

    [JsonPropertyName("url")]
    public string? Url { get; set; }
}
